### PR TITLE
Refactor: podspec

### PR DESCRIPTION
**Pull request by Climbatize**
# Description
Refactor podspec and project structure:

- support for automatic host app versioning based on MiniApp SDK version
- support for subspecs development (for having Admobs as an optional dependency for example)
- added shared schemes to build application in debug mode or release mode to have the ability to generate binaries for QA with crash report activated for example (implemented in another upcoming branch)

# Checklist
- [X] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [X] I have read the [contributing guidelines](CONTRIBUTING.md).
- [X] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data before every commit, including API endpoints and keys
- [X] I ran `fastlane ci` without errors

### CODE REVIEW

File: MiniApp.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -228,7 +232,6 @@
 				38920AD1251D9C57004C5DDD /* Progress+UIViewController.swift */,
 				38920AD2251D9C57004C5DDD /* UIViewController+MiniApp.swift */,
 				38920AD3251D9C57004C5DDD /* MiniApp+UIImage.swift */,
-				18CC83362538172F00EEDB00 /* UIWindow+MiniApp.swift */,
```
Comment (Climbatize) [inline MiniApp.xcodeproj/project.pbxproj:231]: there was a SDK file included directly into the host app

File: MiniApp.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -258,15 +261,6 @@
...
 		};
-		38920ADD251D9C57004C5DDD /* Config */ = {
-			isa = PBXGroup;
-			children = (
-				38920ADE251D9C57004C5DDD /* Config-debug.xcconfig */,
-				38920ADF251D9C57004C5DDD /* Config-release.xcconfig */,
-			);
-			path = Config;
-			sourceTree = "<group>";
-		};
```
Comment (Climbatize) [inline MiniApp.xcodeproj/project.pbxproj:269]: this was a duplicated config folder we could find twice in host app file structure

File: MiniApp.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -771,11 +763,11 @@
 		};
 		607FACF01AFB9204008FA782 /* Debug */ = {
 			isa = XCBuildConfiguration;
+			baseConfigurationReference = 183AF07C24CE642F000ABCBF /* Config-debug.xcconfig */;
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				INFOPLIST_FILE = "$(SRCROOT)/Example/Info.plist";
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
-				MARKETING_VERSION = 2.7.0;
```
Comment (Climbatize) [inline MiniApp.xcodeproj/project.pbxproj:778]: this is now managed by xcconfig file + podspec

File: Example/Config/Config-common.xcconfig [hunk 4]
```diff
@@ -0,0 +1,2 @@
+// IMPORTANT: MiniApp sample host app version is linked to the SDK version defined in the MiniApp.podspec file
+MARKETING_VERSION = $(RMA_SDK_VERSION)$(RMA_DEMO_APP_BUILD_TYPE)
```
Comment (Climbatize) [inline Example/Config/Config-common.xcconfig:2]: this replaces the plist value that uses MARKETING_VERSION, as the developer can accidentally  make the plist come back to the default `$(MARKETING_VERSION)` value

**Review by corycaywood — APPROVED** [review]
LGTM 👍

**Review by ts-leojoseph-r — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**